megamek/src/megamek/server/GameManager.java

```diff
@@ -7359,6 +7359,24 @@ private void processMovement(Entity entity, MovePath md, Map<EntityTargetPair,
                 }
             }
 
+            if (step.getType() == MovePath.MoveStepType.CHAFF) {
+                List<Mounted> chaffDispensers = entity.getMiscEquipment(MiscType.F_CHAFF_POD)
+                        .stream().filter(dispenser -> dispenser.isReady())
+                        .collect(Collectors.toList());
+                if (chaffDispensers.size() > 0) {
+                    chaffDispensers.get(0).setFired(true);
+                    createSmoke(curPos, 1, 1);
+                    Hex hex = game.getBoard().getHex(curPos);
+                    hex.addTerrain(new Terrain(Terrains.SMOKE, SmokeCloud.SMOKE_CHAFF_LIGHT));
+                    sendChangedHex(curPos);
+                    r = new Report(2512)
```

Comment:
There is report 5187; is this applicable here instead of a new one?

Reply:
Different context here is that unit is deploying chaff rather than a grenade simply filling the hex with chaff smoke ?